bcda/bcdacli/cli.go

```diff
@@ -598,19 +610,35 @@ func archiveExpiring(hrThreshold int) error {
 	for _, j := range jobs {
 		t := j.UpdatedAt
 		elapsed := time.Since(t).Hours()
-		if int(elapsed) >= hrThreshold {
-
-			id := int(j.ID)
-			jobDir := fmt.Sprintf("%s/%d", os.Getenv("FHIR_PAYLOAD_DIR"), id)
-			expDir := fmt.Sprintf("%s/%d", os.Getenv("FHIR_ARCHIVE_DIR"), id)
-
-			err = os.Rename(jobDir, expDir)
+		id := int(j.ID)
+		jobDir := fmt.Sprintf("%s/%d", os.Getenv("FHIR_PAYLOAD_DIR"), id)
+		_, err = os.Stat(jobDir)
+		jobDirExist := err == nil
+		expDir := fmt.Sprintf("%s/%d", os.Getenv("FHIR_ARCHIVE_DIR"), id)
+		_, err = os.Stat(expDir)
+		expDirExist := err == nil
+
+		// TODO: Should we care about the threshold when setting archived status to already archived files?
+		if !jobDirExist && expDirExist {
+			log.Info("Archived job files found, setting status to archived...")
+			// TODO: Separate Status update function - might get messy passing db connection.
+			j.Status = models.JobStatusArchived
+			err = db.Save(j).Error
 			if err != nil {
 				log.Error(err)
 				lastJobError = err
-				continue
+			}
+		} else if int(elapsed) >= hrThreshold {
```

Comment:
Instead of filtering out the jobs after we get them back from the DB, we should add an additional clause to our db lookup above. We're doing a similar approach in the `cleanupArchive` method here: https://github.com/CMSgov/bcda-app/blob/b14e1435ee87640bbed303149bdadc43ebbc0b7b/bcda/bcdacli/cli.go#L633

I think if we do it this way, the TODO above is addressed since all jobs satisfy the expiration criteria.

And since the database is handling the filtering, we do not have to check that ` int(elapsed) >= hrThreshold`

Reply:
Matched logic in `cleanupArchive` function and removed the additional conditional in both functions that was always true due to the DB call change.  Tests passing on both and full suite.